vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -755,6 +769,27 @@ def _prepare_decode(
             left_padded_prompt_mask.append(
                 self.req_ids2left_pads[cached_request.req_id])
 
+        # add padding for minimum batch size of 2
+        if len(input_tokens) == 1:
+            dummy_req_indices = torch.zeros(1, dtype=torch.bool, device="cpu")
+            self.model.indices = torch.cat(
+                (self.model.indices, dummy_req_indices), -1)
+            assert self.model.indices.size(dim=0) == 2
+
+            n = self.tkv + 1
+            d = self.BLOCK_SIZE
+            num_blocks = (n + d - 1) // d
+            for i in range(num_blocks - len(self.dummy_req_ids2blocks)):
+                self.dummy_req_ids2blocks.append(self.free_blocks.popleft())
```

Comment:
Is this popping free blocks for every decode iteration of a batch with a single request? It looks like `len(input_tokens) == 1` will always be true when there's only a single request running, since `cached_requests` comes from the scheduler and won't have our dummy request in it

Reply:
ahhh I see the `num_blocks - len(self.dummy_req_ids2blocks)` is only popping ones as the tkv increases 👍